contracts/p1/StRSR.sol

```diff
@@ -329,13 +334,19 @@ abstract contract StRSRP1 is Initializable, ComponentP1, IStRSR, EIP712Upgradeab
         totalDrafts = newTotalDrafts;
         draftRSR = newDraftRSR;
 
+        // == Interactions ==
+        leakyRefresh(rsrAmount);
+        IERC20Upgradeable(address(rsr)).safeTransfer(account, rsrAmount);
         emit UnstakingCompleted(firstId, endId, draftEra, account, rsrAmount);
 
-        // == Interaction ==
-        IERC20Upgradeable(address(rsr)).safeTransfer(account, rsrAmount);
+        // == Checks ==
+        require(basketHandler.fullyCollateralized(), "RToken uncollateralized");
```

Comment:
Why moved this checks to the end? They dont make sense before the refresh right?

Reply:
they're after the refresh, right?

Reply:
Yes, was just asking if that was the reason they were moved to the end. I believe they are.